tests/test_serialization.py

```diff
@@ -811,6 +811,113 @@ def try_serialize(obj):
                     self.assertNotIn("11", deserialized)
 
 
+class TestIncrementalSerialization(unittest.TestCase):
+    # Tests that involve multiple calls to write_tensor()
```

Comment:
It used to be the case that a bulk write could still be incremental, and a series of bulk writes could be used to create a tensorizer file. Bulk write functions are also the only way to incrementally write structured data, namely through `write_state_dict`. As far as I can tell, the way the serialization code is written still allows that, so we should add some test cases for that here too.

Reply:
https://github.com/coreweave/tensorizer/issues/149